--- old/Makefile	2016-03-26 08:18:07.000000000 +0800
+++ new/Makefile	2017-08-01 15:27:38.206230279 +0800
@@ -74,7 +74,7 @@ $(ODIR)/%.o : %.c
 LUAJIT  := $(notdir $(patsubst %.tar.gz,%,$(wildcard deps/LuaJIT*.tar.gz)))
 OPENSSL := $(notdir $(patsubst %.tar.gz,%,$(wildcard deps/openssl*.tar.gz)))
 
-OPENSSL_OPTS = no-shared no-ssl2 no-psk no-srp no-dtls no-idea --prefix=$(abspath $(ODIR))
+OPENSSL_OPTS = no-afalgeng no-shared no-ssl2 no-psk no-srp no-dtls no-idea --prefix=$(abspath $(ODIR))
 
 $(ODIR)/$(LUAJIT):  deps/$(LUAJIT).tar.gz  | $(ODIR)
 	@tar -C $(ODIR) -xf $<
@@ -93,6 +93,7 @@ ifeq ($(TARGET), darwin)
 else
 	@$(SHELL) -c "cd $< && ./config $(OPENSSL_OPTS)"
 endif
+	sed -i 's/^INSTALL_SHLIB_INFO=.*/INSTALL_SHLIB_INFO=/g' obj/openssl-1.0.2g/Makefile
 	@$(MAKE) -C $< depend install
 
 # ------------
