--- boost_1_63_0/boost/function/function_template.hpp.alias	2017-03-24 12:17:21.560125198 +0000
+++ boost_1_63_0/boost/function/function_template.hpp	2017-03-24 13:07:24.971891166 +0000
@@ -994,7 +994,7 @@
         if (!f.empty()) {
           this->vtable = f.vtable;
           if (this->has_trivial_copy_and_destroy())
-            this->functor = f.functor;
+            __builtin_memcpy(&this->functor, &f.functor, sizeof(f.functor));
           else
             get_vtable()->base.manager(f.functor, this->functor,
                                      boost::detail::function::move_functor_tag);
