diff -rup old/SConstruct new/SConstruct
--- old/SConstruct	2017-09-13 10:07:14.873451110 +0800
+++ new/SConstruct	2017-09-13 10:09:10.671507048 +0800
@@ -937,6 +937,13 @@ def set_config_header_define(env, varnam
     env['CONFIG_HEADER_DEFINES'][varname] = varval
 env.AddMethod(set_config_header_define, 'SetConfigHeaderDefine')
 
+env.Replace(CC = '/opt/rh/devtoolset-4/root/usr/bin/gcc')
+env.Replace(CXX = '/opt/rh/devtoolset-4/root/usr/bin/g++')
+env.Replace(CPP = '/opt/rh/devtoolset-4/root/usr/bin/cpp')
+env.Prepend(LIBPATH = ['/opt/rh/devtoolset-4/root/usr/lib64'])
+env.Prepend(CPPPATH = ['/opt/rh/devtoolset-4/root/usr/include'])
+env.Append(RPATH = ['/opt/rh/devtoolset-4/root/usr/lib64'])
+
 detectEnv = env.Clone()
 
 # Identify the toolchain in use. We currently support the following:
@@ -1649,7 +1656,8 @@ if not env.TargetOSIs('windows'):
 
 # boostSuffixList is used when using system boost to select a search sequence
 # for boost libraries.
-boostSuffixList = ["-mt", ""]
+boostSuffixList = [""]
+#boostSuffixList = ["-mt", ""]
 if get_option("system-boost-lib-search-suffixes") is not None:
     if not use_system_version_of_library("boost"):
         env.FatalError("The --system-boost-lib-search-suffixes option is only valid "
